kratos/python_scripts/new_linear_solver_factory.py

```diff
@@ -1,82 +1,8 @@
 from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
 
 
-
-def ConstructPreconditioner(configuration):
-    import KratosMultiphysics
-    if(configuration["preconditioner_type"].GetString() == "None"):
-        return KratosMultiphysics.Preconditioner()
-    elif(configuration["preconditioner_type"].GetString() == "DiagonalPreconditioner"):
-        return KratosMultiphysics.DiagonalPreconditioner()
-    elif(configuration["preconditioner_type"].GetString() == "ILU0Preconditioner"):
-        return KratosMultiphysics.ILU0Preconditioner()
-    elif(configuration["preconditioner_type"].GetString() == "ILUPreconditioner"):
-        return KratosMultiphysics.ILUPreconditioner() 
-    else:
-        raise Exception("preconditioner_type specified is not correct. Current choice is :" + configuration["preconditioner_type"].GetString())
-
 def ConstructSolver(configuration):
     import KratosMultiphysics
+    return KratosMultiphysics.LinearSolverFactoryBase().CreateSolver(configuration)
```

Comment:
Now I'm confused, I think what referred as base is the factory itself and what is referred as factory is a concrete factory.